src/System.Net.Http/src/uap/System/Net/HttpClientHandler.cs

```diff
@@ -684,6 +714,63 @@ private static bool InitRTNoCacheSupported()
                 "NoCache");
         }
 
+        private void RTServerCertificateCallback(RTHttpBaseProtocolFilter sender, RTHttpServerCustomValidationRequestedEventArgs args)
+        {
+            // Convert WinRT certificate to .NET certificate.
+            X509Certificate2 serverCert = ConvertCertificate(args.ServerCertificate);
+
+            // Create .NET X509Chain from the WinRT information. We need to rebuild
+            // the chain since WinRT only gives us an array of intermediate certificates.
+            var serverChain = new X509Chain();
+            foreach (RTCertificate cert in args.ServerIntermediateCertificates)
+            {
+                serverChain.ChainPolicy.ExtraStore.Add(ConvertCertificate(cert));
+            }
+            serverChain.Build(serverCert);
+
+            // Determine chain and name errors from WinRT information.
+            SslPolicyErrors sslPolicyErrors = SslPolicyErrors.None;
+            if (serverChain.ChainStatus.Length > 0)
+            {
+                sslPolicyErrors |= SslPolicyErrors.RemoteCertificateChainErrors;
+            }
+            foreach (RTChainValidationResult result in args.ServerCertificateErrors)
+            {
+                if (result == RTChainValidationResult.InvalidName)
+                {
+                    sslPolicyErrors |= SslPolicyErrors.RemoteCertificateNameMismatch;
+                    break;
+                }
+            }
+
+            // Get the .NET HttpRequestMessage we saved in the property bag of the WinRT HttpRequestMessage.
+            HttpRequestMessage request = (HttpRequestMessage)args.RequestMessage.Properties[RequestMessageLookupKey];
+
+            // Call the .NET callback.
+            bool success = false;
+            try
+            {
+                success = _serverCertificateCustomValidationCallback(request, serverCert, serverChain, sslPolicyErrors);
+            }
+            catch (Exception ex)
+            {
+                // Save the exception. We will return it later via the SendAsync response processing.
+                args.RequestMessage.Properties.Add(SavedExceptionLookupKey, (object)ex);
+            }
+
+            if (!success)
+            {
+                args.Reject();
+            }
+        }
+
+        private X509Certificate2 ConvertCertificate(RTCertificate cert)
```

Comment:
Consider renaming this to "ConvertPublicKeyCertificate" or something to that effect. The conversion being carried out would drop any private keys from the WinRT certificate, which is fine for our needs (server's identity certificates don't contain private keys), but we don't want future developers to assume that this method's conversion approach can be used for all purposes.